File: src/main/java/org/intense/Ast/DisplayNode.java
package org.intense.Ast;

import org.intense.Environment;

public class DisplayNode extends ASTNode {
    public ASTNode getPrintable() {
        return printable;
    }

    public void setPrintable(ASTNode printable) {
        this.printable = printable;
    }

    ASTNode printable = null;

    @Override
    public Object eval(Environment env) {
        System.out.println(printable.eval(null));
        return "()";
    }
}

File: src/main/java/org/intense/ast/DisplayNode.kt
package org.intense.ast;

import org.intense.SymbolTable;
import org.intense.Types.Type

class DisplayNode(mPrintable:ASTNode) : ASTNode() {
    private var printable: ASTNode? = mPrintable
    fun getPrintable() : ASTNode? {
        return printable;
    }

    fun setPrintable(printable: ASTNode) {
        this.printable = printable;
    }

    override fun inferType(env: SymbolTable): Type {
        TODO("Not yet implemented")
    }

    override fun toKotlinCode(env: SymbolTable): String {
        return "println "+printable?.toKotlinCode(env)
    }
}
